src/server/raft_client.rs

```diff
@@ -243,6 +237,22 @@ impl<T: RaftStoreRouter> RaftClient<T> {
     }
 }
 
+// Collect raft messages into a vector so that we can merge them into one message later.
+// `MAX_GRPC_SEND_MSG_LEN` will be considered when collecting.
+struct RaftMsgCollector(usize);
+impl BatchCollector<Vec<RaftMessage>, RaftMessage> for RaftMsgCollector {
+    fn collect(&mut self, v: &mut Vec<RaftMessage>, e: RaftMessage) -> Option<RaftMessage> {
+        let msg_size = e.compute_size() as usize;
```

Comment:
How about the cost of `compute_size`?

Reply:
I worry about `compute_size` performance. The implementation is looping through **all** elements recursively. Maybe we should do a benchmark.

Reply:
How about only care size of `Entry`?